kafka/src/test/groovy/io/micronaut/configuration/kafka/errors/KafkaErrorStrategySpec.groovy

```diff
@@ -70,6 +70,22 @@ class KafkaErrorStrategySpec extends AbstractEmbeddedServerSpec {
         myConsumer.times[1] - myConsumer.times[0] >= 50
     }
 
+    void "test when the error strategy is 'retry on error' and there are serialization errors"() {
+        when:"A record cannot be deserialized"
+        DeserializationErrorClient myClient = context.getBean(DeserializationErrorClient)
+        myClient.sendText("Not an integer")
+        myClient.sendNumber(123)
+
+        RetryOnErrorDeserializationErrorConsumer myConsumer = context.getBean(RetryOnErrorDeserializationErrorConsumer)
+
+        then:"The message that threw the exception is eventually left behind"
+        conditions.eventually {
+            myConsumer.number == 123
+        }
+        and:"the retry error strategy is honored"
+        myConsumer.exceptionCount.get() == 2
```

Comment:
should not this be 1 since there has been only one error when we sent once `Not an integer`?

Reply:
Nope, there's been 2 errors:
- First exception is thrown the first time we try to deserialize the message.
- Then [the error strategy kicks in with `DEFAULT_RETRY_COUNT=1`](https://github.com/micronaut-projects/micronaut-kafka/pull/771/files/aa0d66fe89c9d208184cc47ac155f1c5bbfa76c5#diff-6b0f4620784fab078897bdfde65e1ce637a9d85668ceff6799b1739880c98fbeR321), so we try to deserialize it once more and the second exception is thrown.

All the exceptions are delivered to the handler.